README.md

```diff
@@ -173,6 +173,64 @@ console.log(await promptWithCalculator("What is 2 + 2?"));
 
 We'll likely explore more specific APIs for tool- and function-calling in the future; follow along in [issue #7](https://github.com/webmachinelearning/prompt-api/issues/7).
 
+### Multimodal inputs
+
+All of the above examples have been of text prompts. Some language models also support other inputs. Our design initially includes the potential to support images and audio clips as inputs. This is done by using objects in the form `{ type: "image", data }` and `{ type: "audio", data }` instead of strings. The `data` values can be the following:
+
+* For image inputs: [`ImageBitmapSource`](https://html.spec.whatwg.org/#imagebitmapsource), i.e. `Blob`, `ImageData`, `ImageBitmap`, `VideoFrame`, `OffscreenCanvas`, `HTMLImageElement`, `SVGImageElement`, `HTMLCanvasElement`, or `HTMLVideoElement` (will get the current frame). Also raw bytes via `BufferSource` (i.e. `ArrayBuffer` or typed arrays).
+
+* For audio inputs: for now, `Blob`, `AudioBuffer`, `HTMLAudioElement`. Also raw bytes via `BufferSource`. Other possibilities we're investigating include `AudioData` and `MediaStream`, but we're not yet sure if those are suitable to represent "clips".
+
+Sessions that will include these inputs need to be created using the `expectedInputTypes` option, to ensure that any necessary downloads are done as part of session creation, and that if the model is not capable of such multimodal prompts, the session creation fails.
+
+A sample of using these APIs:
+
+```js
+const session = await ai.languageModel.create({
+  expectedInputTypes: ["audio", "image"] // "text" is always expected
+});
+
+const referenceImage = await (await fetch("/reference-image.jpeg")).blob();
+const userDrawnImage = document.querySelector("canvas");
+
+const response1 = await session.prompt([
```

Comment:
I think passing the prompt as the first parameter of an array doesn't appeal to me as a developer. I would personally prefer an array of objects that each have their own label (code is more readable and when reading the documentation, it's easier to understand what to pass where). I think this is due to the first parameter being a `AILanguageModelPromptInput` right?

What about something like 

```
session.prompt( [{
    input: AILanguageModelPromptInput,
    media: AILanguageModelPromptMedia
}]
```

`AILanguageModelPromptMedia` could be something like (forgive me this is in TypeScript-ish) : 

```
type AILanguageModelPromptMedia {
   type: "audio" | "image",
   data: Blob
}
```

Reply:
IIUC, what you're hoping for is mostly already possible, IDL union allowances permit strings or structs or an array of mixed strings or structs. So these are all possible:
```
session.prompt([
  { role: 'system', content: { type: 'text', data: 'my string' }},
  { role: 'user', content: { type: 'image', data: myImage }},
]);

session.prompt([
  { type: 'text', data: 'my string' },
  { type: 'image', data: myImage },
]);

session.prompt([
  'my string',
  { type: 'image', data: myImage },
]);

session.prompt('my string']);
```

Reply:
Yes, I think this is all possible (and a bit simpler now that the dictionaries are flattened). I'll resolve this, but let me know if we've misunderstood!